packages/frontend/src/Build.spec.ts

```diff
@@ -0,0 +1,142 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { vi, test, expect } from 'vitest';
+import { screen, render } from '@testing-library/svelte';
+import Build from './Build.svelte';
+import type { BootcBuildInfo } from '@shared/src/models/bootc';
+import type { ImageInfo } from '@podman-desktop/api';
+
+const mockHistoryInfo: BootcBuildInfo[] = [
+  {
+    name: 'image1',
+    engineId: 'engine1',
+    tag: 'latest',
+    type: 'iso',
+    folder: '/tmp/image1',
+    arch: 'x86_64',
+  },
+  {
+    name: 'image2',
+    engineId: 'engine2',
+    tag: 'latest',
+    type: 'iso',
+    folder: '/tmp/image1',
+    arch: 'x86_64',
+  },
+];
+
+// Mocked bootc images, with one containing the 'bootc' and 'containers.bootc' labels, and the other not
+const mockBootcImages: ImageInfo[] = [
+  {
+    Id: 'image1',
+    RepoTags: ['image1:latest'],
+    Labels: {
+      bootc: 'true',
+    },
+    engineId: 'engine1',
+    engineName: 'engine1',
+    ParentId: 'parent1',
+    Created: 0,
+    VirtualSize: 0,
+    Size: 0,
+    Containers: 0,
+    SharedSize: 0,
+  },
+  {
+    Id: 'image2',
+    RepoTags: ['image2:latest'],
+    Labels: {
+      bootc: 'true',
+    },
+    engineId: 'engine2',
+    engineName: 'engine2',
+    ParentId: 'parent2',
+    Created: 0,
+    VirtualSize: 0,
+    Size: 0,
+    Containers: 0,
+    SharedSize: 0,
+  },
+];
+
+const mocks = vi.hoisted(() => {
+  return {
+    listHistoryInfo: vi.fn(),
+    listBootcImages: vi.fn(),
+  };
+});
```

Comment:
we don't need this hoisted block

Reply:
If we do not use it, we get this error:

```
 FAIL  src/Build.spec.ts [ src/Build.spec.ts ]
Error: [vitest] There was an error when mocking a module. If you are using "vi.mock" factory, make sure there are no top level variables inside, since this call is hoisted to top of the file. Read more: https://vitest.dev/api/vi.html#vi-mock
Caused by: ReferenceError: Cannot access 'listHistoryInfoMock' before initialization
 ❯ src/Build.spec.ts:5:24
 ❯ src/Build.svelte:6:31
```

When trying to "mock" it our usual way.

So had to use hoisted: https://vitest.dev/api/vi.html#vi-mock

Reply:
I think you need to use the comments below

you don't need at all `listHistoryInfoMock`

Reply:
tried that and unfortunately still ran into the same error. I believe it's because we mock `./api/client` ?

Reply:
this is something that aistudio ran into issues too, see here: https://github.com/projectatomic/ai-studio/blob/main/packages/backend/src/studio-api-impl.spec.ts#L54

Reply:
https://github.com/cdrage/podman-desktop-extension-bootc/pull/1

Reply:
here is PR that is removing hoisted usage and simplify it and brings type safety

Reply:
Ahhh, that's what I was doing wrong. I was creating a `const mockBootcListImages = vi.fn()` as well, and using that. By importing bootcapi, it fixes it. My mistake!